### PR TITLE
Paddle frontend: mse_loss #17400

### PR DESCRIPTION
**Pull request by Kamil2601**
Close - [ ] #17400

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by MuhammedAshraf2020** [comment]
Hey @Kamil2601, thanks a lot for the PR. Some minor changes and we are good to go with the final tests

File: ivy/functional/frontends/paddle/nn/functional/loss.py [hunk 1]
```diff
@@ -43,3 +43,12 @@ def binary_cross_entropy_with_logits(
...
     return paddle.to_tensor(ret.reshape([-1]))
+
+
+@with_supported_dtypes({"2.4.2 and below": ("float32", "float64")}, "paddle")
+@inputs_to_ivy_arrays
+def mse_loss(input, label, reduction="mean", name=None):
+    reduction = _get_reduction_func(reduction)
+    ret = ivy.square(input - label)
+    ret = reduction(ret)
+    return paddle.to_tensor(ret.reshape([-1]))
```
Comment (MuhammedAshraf2020) [inline ivy/functional/frontends/paddle/nn/functional/loss.py:54]: It appears that the flatten is causing unexpected behavior compared to the original function when the reduction is set to `none`. In this case, I think there is no need for the flatten operation.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_loss.py [hunk 2]
```diff
@@ -56,3 +56,59 @@ def test_paddle_binary_cross_entropy_with_logits(
...
+        min_value=0.0,
+        max_value=1.0,
+        large_abs_safety_factor=2,
+        small_abs_safety_factor=2,
+        safety_factor_scale="linear",
+        allow_inf=False,
+        exclude_min=True,
+        exclude_max=True,
+        min_num_dims=1,
+        max_num_dims=1,
```
Comment (MuhammedAshraf2020) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_loss.py:75]: I think you are not covering all of the input space

**Review by MuhammedAshraf2020 — CHANGES_REQUESTED** [review]

**Comment by Kamil2601** [comment]
@MuhammedAshraf2020 fixed

**Comment by MuhammedAshraf2020** [comment]
Hey @Kamil2601, As you see here, the frontend function doesn't act similar to the original function with the same input shape/dtype 

Screenshot from 2023-06-27 16-47-07

**Comment by Kamil2601** [comment]
@MuhammedAshraf2020 the problem is with binary_cross_entropy_loss, not mse
image

For me original paddle mse_loss returns tensor of shape (1,)
image

**Comment by MuhammedAshraf2020** [comment]
It seems that there is a problem with different versions, we will have a look on that

**Review by MuhammedAshraf2020 — APPROVED** [review]

**Comment by MuhammedAshraf2020** [comment]
LGTM
